server/src/main/java/io/crate/metadata/upgrade/MetadataIndexUpgrader.java

```diff
@@ -58,6 +62,8 @@ public IndexMetadata apply(IndexMetadata indexMetadata,
      */
     private IndexMetadata createUpdatedIndexMetadata(IndexMetadata indexMetadata, @Nullable IndexTemplateMetadata indexTemplateMetadata) {
         return IndexMetadata.builder(indexMetadata)
+            // UUID is not re-used what are the consequences?
+            .index(fixIndexName(indexMetadata.getIndex().getName()))
```

Comment:
The UUID will be reused due to the `indexMetadata` which is passed to the builder above and such the index `settings` are reused, which contain the UUID. Otherwise it probably wouldn't work as expected as the index data path contains the UUID (with a new UUID, old data would be lost).